ldelf/ta_elf.c

```diff
@@ -261,10 +281,21 @@ static void e64_save_symtab(struct ta_elf *elf, size_t tab_idx)
 
 	elf->dynsymtab = (void *)(vaddr_t)(shdr[tab_idx].sh_addr +
 					   elf->load_addr);
-	assert(!(shdr[tab_idx].sh_size % sizeof(Elf64_Sym)));
+
+	if (!ALIGNMENT_IS_OK(elf->dynsymtab, Elf64_Sym))
+		err(TEE_ERROR_GENERIC, "Bad alignment of dynsymtab %p",
+		    elf->dynsymtab);
+	check_range(elf, "dynsymtab", elf->dynsymtab, shdr[tab_idx].sh_size);
+
+	if (shdr[tab_idx].sh_size % sizeof(Elf64_Sym))
+		err(TEE_ERROR_GENERIC,
+		    "Size of dynsymtab not an even multiple of Elf64_Sym");
```

Comment:
very minor: also maybe over a single line (unless Elf64_Sym/Elf32_Sym reference is useful)

Reply:
I'd prefer not making the debug printout more cryptic that necessary.